### PR TITLE
Set route to metadata on interface with default route

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: google_guest_agent/instance_setup.go [hunk 1]
```diff
@@ -58,44 +57,48 @@ func agentInit(ctx context.Context) error {
...
-			return err
+			logger.Errorf("%s, error listing IPForwardEntries: %v", msg, err)
+			return
 		}
 
-		interfaces, err := net.Interfaces()
-		if err != nil {
-			return err
+		// Choose the index that has the default route setup.
+		// This is equivalent to
```
[the remark this answers is not in the document]
Comment (adjackura) [reply google_guest_agent/instance_setup.go:68]: I apparently sent this out before I was done

File: google_guest_agent/instance_setup.go [hunk 2]
```diff
@@ -58,44 +57,48 @@ func agentInit(ctx context.Context) error {
...
-				ipForwardMetric1: 1,
-				ipForwardIfIndex: int32(iface.Index),
-			}
+		iface, err := net.InterfaceByIndex(int(index))
+		if err != nil {
+			logger.Errorf("%s, error from net.InterfaceByIndex(%d): %v", msg, index, err)
+			return
+		}
 
-			if forwardEntryExists(fes, fe) {
```
Comment (hopkiw) [inline google_guest_agent/instance_setup.go:90]: if this isn't needed, delete the function?

**Pull request merged**